### ISSUE TITLE
Output mapping inside start events maps the wrong variable values in parallel multi embedded sub processes

### ISSUE THREAD

**Issue by allanbarklie** [camunda-cloud/zeebe#7072]
**Describe the bug**
A workflow includes a parallel multi embedded sub process.
The start event for the sub process is configured with an out parameter map.
The input for the map is the instance specific value from the multi -> specific instance transform and the output is another new variable.

So if a multi is passed "messages": ["A", "B"] and configured to map from collection =messages to input element messageInnerStart then separately the Start event for the sub process is configured to map from messageInnerStart  to message.

Then the value stored in message does not always match the expected value from messages.
(messageInnerStart   is reported correctly so the error seems to occur going from messageInnerStart   to message)

This can be worked around by avoiding using mappings on the start event - direct access of messageInnerStart   from later stages works correctly.

**To Reproduce**
Please see simple attached workflow.
[multi-sub-process-test.bpmn.txt](https://github.com/camunda-cloud/zeebe/files/6503433/multi-sub-process-test.bpmn.txt)

Rename the attached file to sub-process-test.bpmn

`./zbctl --insecure deploy multi-sub-process-test.bpmn`

In a separate shell window
`./zbctl --insecure create worker print-message --handler cat`

In the original shell
`./zbctl --insecure create instance multi-sub-process-test --variables '{"correlationValue":1, "messages": ["A", "B"]}'`

**Expected behavior**

I expect the values to map through so that "message" gets the value "A" in one instance and "B" in the other.

Instead I get this:
```
2021/05/18 19:24:17 Activated job 2251799813728647 with variables {"loopCounter":2,"messageInnerStart":"B","message":"B","messages":["A","B"],"correlationValue":1}
2021/05/18 19:24:17 Activated job 2251799813728646 with variables {"loopCounter":1,"messageInnerStart":"A","message":"B","messages":["A","B"],"correlationValue":1}
2021/05/18 19:24:17 Handler completed job 2251799813728646 with variables {"loopCounter":1,"messageInnerStart":"A","message":"B","messages":["A","B"],"correlationValue":1}
2021/05/18 19:24:17 Handler completed job 2251799813728647 with variables {"loopCounter":2,"messageInnerStart":"B","message":"B","messages":["A","B"],"correlationValue":1}
```

"messageInnerstart" has the expected values but "message" is set to "B" in both instances

**Environment:**
 - OS: WSL2 (Centos8)
 - Zeebe Version: 0.26.0
 - Configuration: hazelcast/simple monitor, single broker with no separate gateway

**Comment by saig0** [comment]
@allanbarklie thanks for raising this up. The described behavior is **expected**.

In the example, the variable `message` is overridden by the second inner instance of the multi-instance body before the logging task is executed. This happens because the variable is defined in the (root) scope of the process instance. 

It can be solved by defining an input mapping on the **multi-instance embedded subprocess** (instead of the start event):

```
message

= messageInnerStart
```

Does this help you?

Please have a look at the docs: https://docs.camunda.io/docs/reference/bpmn-processes/multi-instance/multi-instance#variable-mappings

**Comment by npepinpe** [comment]
Same as the other issue, please create a docs issue and link it here, and close this issue.

**Comment by saig0** [comment]
I created a new issue for the docs: camunda-cloud/camunda-cloud-documentation#263

Please reopen the issue if the actual issue is not solved by the provided fix.
